airflow_prometheus_exporter/prometheus_exporter.py

```diff
@@ -407,22 +439,45 @@ def sla_check(sla_interval, sla_time, max_execution_date, cadence, execution_dat
 def get_sla_miss_dags():
     min_date_to_filter = pendulum.now(TIMEZONE).subtract(days=RETENTION_TIME)
     with session_scope(Session) as session:
-        execution_dt_query = (
+        active_dag_query = (
+            session.query(
+                DelayAlertMetaData.dag_id
+            )
+            .join(DagModel, DelayAlertMetaData.dag_id == DagModel.dag_id)
+            .filter(
+                DelayAlertMetaData.task_id.is_(None),
+                DagModel.is_active == True,
+                DagModel.is_paused == False,
+            )
+            .group_by(DelayAlertMetaData.dag_id)
+            .subquery()
+        )
+
+        execution_date_query = (
             session.query(
                 DagRun.dag_id,
-                DagModel.schedule_interval,
                 DagRun.execution_date,
                 DagRun.state,
             )
-            .join(DagModel, DagModel.dag_id == DagRun.dag_id)
+            .join(active_dag_query, active_dag_query.c.dag_id == DagRun.dag_id)
             .filter(
-                DagModel.is_active == True,
-                DagModel.is_paused == False,
                 DagRun.execution_date > min_date_to_filter,
             )
             .order_by(DagRun.execution_date.desc())
-            .subquery()
+            .all()
         )
+
+        execution_dates = {}
+        max_execution_dates = {}
+        for r in execution_date_query:
+            key = r.dag_id
+            if key not in execution_dates:
+                execution_dates[key] = []
```

Comment:
unnecessary but if you want to take advantage of the collections library, you could turn this code into
```
from collections import defaultdict

execution_dates = defaultdict(list)
for r in execution_date_query:
    execution_dates[r.dag_id].append(r.execution_date)
```